File: converted/classes_only/FileSystemImpl$CloseTask.java
package zombie.fileSystem;

final class FileSystemImpl$CloseTask extends FileTask {
   IFile m_file;
   IFileTask2Callback m_cb;

   FileSystemImpl$CloseTask(FileSystem var1) {
      super(var1);
   }

   public Object call() throws Exception {
      this.m_file.close();
      this.m_file.release();
      return null;
   }

   public void handleResult(Object var1) {
      if (this.m_cb != null) {
         this.m_cb.onFileTaskFinished(this.m_file, var1);
      }
   }

   public void done() {
      this.m_file = null;
      this.m_cb = null;
   }
}

File: ProjectZomboidSwitch/src/zombie/fileSystem/FileSystemImpl$CloseTask.cpp
#include "zombie/fileSystem/FileSystemImpl$CloseTask.h"

namespace zombie {
namespace fileSystem {

void* FileSystemImpl$CloseTask::call() {
    m_file->close();
    m_file->release();
    return nullptr;
}
}

void FileSystemImpl$CloseTask::handleResult(void* var1) {
    if (m_cb) {
        m_cb->onFileTaskFinished(m_file, var1);
    }
}
}

void FileSystemImpl$CloseTask::done() {
    m_file = nullptr;
    m_cb = nullptr;
}
}

} // namespace fileSystem
} // namespace zombie
